src/repository/views.py

```diff
@@ -207,50 +208,48 @@ def repository_list(request, subject_slug=None):
 
 
 # TODO: Re-implement
-def preprints_search(request, search_term=None):
+def repository_search(request, search_term=None):
     """
-    Searches through preprints based on their titles and authors
+    Searches through repository content based on titles and authors
     :param request: HttpRequest
     :param search_term: Optional string
     :return: HttpResponse
     """
     if search_term:
         split_search_term = search_term.split(' ')
 
-        article_search = submission_models.Article.preprints.filter(
+        repository_search = models.Preprint.objects.filter(
             (Q(title__icontains=search_term) |
-             Q(subtitle__icontains=search_term) |
+             Q(abstract__icontains=search_term) |
              Q(keywords__word__in=split_search_term)),
-            stage=submission_models.STAGE_PREPRINT_PUBLISHED, date_published__lte=timezone.now()
+            stage=models.STAGE_PREPRINT_PUBLISHED, date_published__lte=timezone.now()
         )
-        article_search = [article for article in article_search]
+        repository_search = [preprint for preprint in repository_search]
 
-        institution_query = reduce(operator.and_, (Q(institution__icontains=x) for x in split_search_term))
+        # institution_query = reduce(operator.and_, (Q(institution__icontains=x) for x in split_search_term))
 
-        from_author = core_models.Account.objects.filter(
-            (Q(first_name__in=split_search_term) |
-             Q(last_name__in=split_search_term) |
-             institution_query)
-        )
+        # from_author = models.Preprint.objects.filter(
+        #     (Q(preprintauthor__in=split_search_term))
+        # )
 
-        articles_from_author = [article for article in submission_models.Article.preprints.filter(
-            authors__in=from_author,
-            stage=submission_models.STAGE_PREPRINT_PUBLISHED,
-            date_published__lte=timezone.now())]
+        # preprints_from_author = [preprint for preprint in models.Preprint.objects.filter(
```

Comment:
Oops... I forgot that I'd commented the author and institution search to get this working... that remains to be done.

Reply:
I think I'm in over my head with this... I can't seem to make progress on searching by author name or institution... we may want to defer that functionality for a future ticket?